syntax = "proto3";

package todo;

import "google/protobuf/empty.proto";

service TodoService {
  rpc Create(TodoRequest) returns (Todo) {}
  rpc Read(TodoRequest) returns (Todo) {}
  rpc Update(TodoRequest) returns (Todo) {}
  rpc Delete(TodoRequest) returns (google.protobuf.Empty) {}
  rpc List(google.protobuf.Empty) returns (TodoListResponse) {}
}
message Todo {
  int64 id = 1;
  string title = 2;
  string description = 3;
  bool is_completed = 4;
  bool is_important=5;
}

message TodoListResponse {
  repeated Todo todos = 1;
}

message TodoRequest {
  int64 id = 1;
  string title = 2;
  string description = 3;
  bool is_completed = 4;
  bool is_important=5;
}

